### PR TITLE
add components.xml, pack it into jar and load it from there correctly

### PR DESCRIPTION
**Pull request by 0x203**
Does anything speak against packing a components.xml as default with this?

Before this, there is a default path specified, but not file in the distributed jar. This pull request adds a components.xml into the resources directory and processes it correctly, even if its read from within a jar-file.

Others can still provide a filename to this library to use a custom components.xml file.

### CODE REVIEW

**Comment by nkluge** [comment]
Packaging a components.xml by default in common.jar is a good idea.

I have to check if the other tools (with custom components.xml files packaged in their jar files) would still work.

File: src/main/java/de/uni_potsdam/hpi/asg/common/breeze/model/xml/Components.java [hunk 1]
```diff
@@ -51,8 +51,8 @@ public static Components readIn(String filename) {
 			JAXBContext jaxbContext = JAXBContext.newInstance(Components.class);
 			Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
 			if(filename == null || filename.equals("")) {
-				URL url = Components.class.getResource(injarfilename);
```
Comment (nkluge) [inline src/main/java/de/uni_potsdam/hpi/asg/common/breeze/model/xml/Components.java:54]: What is the problem with this code? This works for me. However your solution also works. I'm just curious.

Comment (0x203) [inline src/main/java/de/uni_potsdam/hpi/asg/common/breeze/model/xml/Components.java:54]: I've read somewhere that the stream version works the same way no matter if the file is packed in a jar or not. Some said the URL version does not work or at least not as stable, although i didn't verified this.

Comment (0x203) [inline src/main/java/de/uni_potsdam/hpi/asg/common/breeze/model/xml/Components.java:54]: This answer describes it, although I'm not sure if this only applies if one wants to create a File from the URL/URI:
http://stackoverflow.com/a/20389418/2148890

**Pull request merged**